server/app/services/applicant/ApplicantService.java

```diff
@@ -465,6 +469,175 @@ public ImmutableList<Application> getApplications(TimeFilter submitTimeFilter) {
+                    maybeDraftApp.get().getProgram().getProgramDefinition(),
+                    latestSubmittedApplicationTime));
+            programNamesWithApplications.add(programName);
+          } else if (maybeSubmittedApp.isPresent() && activeProgramNames.containsKey(programName)) {
+            ProgramDefinition programDef = activeProgramNames.get(programName);
+            submittedPrograms.add(
+                ApplicantProgramData.create(programDef, latestSubmittedApplicationTime));
+            programNamesWithApplications.add(programName);
+          }
+        });
+
+    Set<String> unappliedActivePrograms =
+        Sets.difference(activeProgramNames.keySet(), programNamesWithApplications);
+    unappliedActivePrograms.forEach(
+        programName -> {
+          unappliedPrograms.add(
+              ApplicantProgramData.create(activeProgramNames.get(programName), Optional.empty()));
+        });
+
+    // Ensure each list is ordered by database ID for consistent ordering.
+    return RelevantPrograms.builder()
+        .setInProgress(sortByProgramId(inProgressPrograms.build()))
+        .setSubmitted(sortByProgramId(submittedPrograms.build()))
+        .setUnapplied(sortByProgramId(unappliedPrograms.build()))
+        .build();
+  }
+
+  private ImmutableList<ApplicantProgramData> sortByProgramId(
+      ImmutableList<ApplicantProgramData> programs) {
+    return programs.stream()
+        .sorted(Comparator.comparing(p -> p.program().id()))
+        .collect(ImmutableList.toImmutableList());
+  }
+
+  private void checkForDuplicateDrafts(ImmutableSet<Application> applications) {
+    // Add logging to better understand a bug where some applicants were seeing
+    // duplicates of programs for which they had draft applications. We can remove
+    // this logging once we determine and resolve the root cause of the duplicate
+    // draft applications.
+    Collection<Map<LifecycleStage, List<Application>>> groupedByStatus =
+        applications.stream()
+            .collect(
+                Collectors.groupingBy(
+                    a -> {
+                      return a.getProgram().getProgramDefinition().adminName();
+                    },
+                    Collectors.groupingBy(Application::getLifecycleStage)))
+            .values();
+    for (Map<LifecycleStage, List<Application>> programAppsMap : groupedByStatus) {
+      List<Application> draftApplications =
+          programAppsMap.getOrDefault(LifecycleStage.DRAFT, Lists.newArrayList());
+      if (draftApplications.size() > 1) {
+        String joinedProgramIds =
+            String.join(
+                ", ",
+                draftApplications.stream()
+                    .map(a -> String.format("%d", a.getProgram().getProgramDefinition().id()))
+                    .collect(ImmutableList.toImmutableList()));
+        logger.debug(
+            String.format(
+                "DEBUG LOG ID: 98afa07855eb8e69338b5af13236a6b7. Program"
```

Comment:
what is this random ID?  It's not clear it's utility or why is needs to be SUPER unique rather than search-filter unique.

Reply:
It's copied forward from where it was originally in ApplicantProgramsController. I'm not planning to change it.